pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -303,34 +512,43 @@ impl<'a> VectoredBlobReader<'a> {
             read.size(),
             buf.capacity()
         );
+
+        if cfg!(debug_assertions) {
+            let align = virtual_file::get_io_buffer_alignment() as u64;
+            debug_assert_eq!(
+                read.start % align,
+                0,
+                "Read start at {} does not satisfy the required io buffer alignment ({} bytes)",
+                read.start,
+                align
+            );
+        }
+
         let mut buf = self
             .file
             .read_exact_at(buf.slice(0..read.size()), read.start, ctx)
             .await?
             .into_inner();
 
         let blobs_at = read.blobs_at.as_slice();
-        let start_offset = blobs_at.first().expect("VectoredRead is never empty").0;
 
-        let mut metas = Vec::with_capacity(blobs_at.len());
+        // Note: read.start != blobs_at.first().start
```

Comment:
Don't think this comment is helpful down the line.

Reply:
removed in fe2393c8c3a4ae37890f87f9d767489f4acf51af